etc/systemd/system/zfs-load-key.service.in

```diff
@@ -0,0 +1,20 @@
+# Based on systemd-sysv-generator output of /etc/init.d/zfs-load-key
+[Unit]
+SourcePath=/etc/init.d/zfs-load-key
+Description=Load ZFS keys for filesystems and volumes
+Documentation=man:zfs-load-key(8)
+Before=zfs-mount.service
+After=zfs-import.service
+DefaultDependencies=no
+
+[Service]
+Type=forking
+Restart=no
+TimeoutSec=5min
+IgnoreSIGPIPE=no
+KillMode=process
+GuessMainPID=no
+RemainAfterExit=yes
+SuccessExitStatus=5 6
+ExecStart=/etc/init.d/zfs-load-key start
```

Comment:
I have little experience with systemd. What makes me suspicious is the dependency on the init script. I guess that's not a nice way. Better would be to have a more generic systemd unit.

Reply:
Without this file `systemd-sysv-generator` will generate this service from init.d:
```
# systemctl cat zfs-load-key
# /run/systemd/generator.late/zfs-load-key.service
# Automatically generated by systemd-sysv-generator

[Unit]
Documentation=man:systemd-sysv-generator(8)
SourcePath=/etc/init.d/zfs-load-key
Description=LSB: Load ZFS keys for filesystems and volumes 22
Before=multi-user.target
Before=multi-user.target
Before=multi-user.target
Before=graphical.target
Before=zfs-mount.service
Before=basic.service
After=zfs-import.service

[Service]
Type=forking
Restart=no
TimeoutSec=5min
IgnoreSIGPIPE=no
KillMode=process
GuessMainPID=no
RemainAfterExit=yes
SuccessExitStatus=5 6
ExecStart=/etc/init.d/zfs-load-key start
ExecStop=/etc/init.d/zfs-load-key stop
```

While it's great not to depend on init.d, I prefer to have one place for code and maybe later we can refactor it. I think it would be great to fix the problem before that.

Reply:
To be precise - service file in this PR is nearly the same as from systemd-sysv-generator, but with `DefaultDependencies=no` and minor cleanups.

Reply:
This may work with the Debian bookworm. But maybe doesn't work on other Linux distributions or even breaks them. The provided systemd units of OpenZFS should be generic.